### ISSUE TITLE
Did you use non linearity in your code?

### ISSUE THREAD

**Issue by sunshineatnoon** [yaolubrain/cnn_linear_max#1]
I noticed this code:
o1 = zeros(convDim1,convDim1,filterNum1,mbSize);  
        for i = 1:filterNum1
            o1(:,:,i,:) = convn(mbImages,rot90(Wc1(:,:,i),2),'valid') + bc1(i);  
        end 
it seems there is no sigmoid function or other non-linearity functions?

**Comment by yaolubrain** [comment]
No. That's why this project is called cnn_linear(neuron)_max(max pooling) :)

On Sat, Jun 13, 2015 at 5:59 AM, SunshineAtNoon notifications@github.com
wrote:

> I noticed this code:
> o1 = zeros(convDim1,convDim1,filterNum1,mbSize);
> 
> for i = 1:filterNum1
> o1(:,:,i,:) = convn(mbImages,rot90(Wc1(:,:,i),2),'valid') + bc1(i);
> 
> end
> it seems there is no sigmoid function or other non-linearity functions?
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/yaolubrain/cnn_linear_max/issues/1.

## 

Yao Lu
Department of Computer Science,
University of Helsinki, Finland
http://yaolubrain.github.io/

**Comment by sunshineatnoon** [comment]
I got it , Thanks. But do few linear layers actually equals to one linear layer?

**Comment by yaolubrain** [comment]
No. Max pooling is a nonlinear function.
